models/repo/repo.go

```diff
@@ -672,6 +672,49 @@ func GetRepositoryByName(ownerID int64, name string) (*Repository, error) {
 	return repo, err
 }
 
+// getRepositoryURLPathSegments returns segments (owner, reponame) extracted from a url
+func getRepositoryURLPathSegments(repoURL string) []string {
+	if strings.HasPrefix(repoURL, setting.AppURL) {
```

Comment:
AppSubURL

Reply:
It's need to be `AppURL` here since `repoUrl` can start with `https://my.domain.tld` which we also need to get rid of.

If this should be changed, we would need to check/remove the protocol & domain seperatly beforehand.

Reply:
repoUrl maybe `https://my.domain.tld/subpath/owner/repo`?

Reply:
Yes; this method was moved from an anonymous inner function of GetRepositoryByURL which is used to determine the repo by its (git clone) url.

It needs to handle these cases:
```
https://my.domain/sub-path/<owner>/<repo>.git
https://my.domain/sub-path/<owner>/<repo>
git+ssh://user@my.domain/<owner>/<repo>.git
git+ssh://user@my.domain/<owner>/<repo>
user@my.domain:<owner>/<repo>.git
user@my.domain:<owner>/<repo>
```